src/mysql-util/src/decoding.rs

```diff
@@ -174,6 +174,35 @@ fn val_to_datum<'a>(
                         temp_strs.push(val.to_string());
                         Datum::from(temp_strs.last().unwrap().as_str())
                     }
+                    Some(MySqlColumnMeta::Date) => {
+                        // Some MySQL dates are invalid in chrono/NaiveDate (e.g. 0000-00-00), so
+                        // we need to handle them directly as strings
+                        if let Value::Date(y, m, d, 0, 0, 0, 0) = value {
+                            temp_strs.push(format!("{:04}-{:02}-{:02}", y, m, d));
```

Comment:
Just double checking that this is the _exact_ textual form used by MySQL. I couldn't find any examples of MySQL not following this format—e.g., negative years aren't supported—but curious how deeply you looked into this.

Reply:
based on these docs these are the format https://dev.mysql.com/doc/refman/8.0/en/datetime.html but that's about as far as I looked too